Lib/test/test_random.py

```diff
@@ -431,14 +431,40 @@ def test_bug_27706(self):
              '0x1.aeaa51052e978p-2', '0x1.092178fb945a6p-2'])
 
     @support.cpython_only
-    def test_issue31478(self):
-        # There shouldn't be an assertion failure in seed() in case the 'a'
-        # argument has a bad __abs__() method.
+    def test_bug_31478(self):
+        # There shouldn't be an assertion failure in _random.Random.seed() in
+        # case the argument has a bad __abs__() method.
         class BadInt(int):
             def __abs__(self):
                 return None
-        with self.assertRaises(TypeError):
-            self.gen.seed(BadInt())
+        self.gen.seed(BadInt())
```

Comment:
I think that if wrap this with `try: ... except TypeError: pass`, the `cpython_only` decorator can be removed.